File: windows/Liferay.SDK/Service/V72/LayoutSet/LayoutSetService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V72.LayoutSet
{
	public class LayoutSetService : ServiceBase
	{
		public LayoutSetService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> UpdateLookAndFeelAsync(long groupId, bool privateLayout, string themeId, string colorSchemeId, string css)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("privateLayout", privateLayout);
			_parameters.Add("themeId", themeId);
			_parameters.Add("colorSchemeId", colorSchemeId);
			_parameters.Add("css", css);

			var _command = new JsonObject()
			{
				{ "/layoutset/update-look-and-feel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateSettingsAsync(long groupId, bool privateLayout, string settings)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("privateLayout", privateLayout);
			_parameters.Add("settings", settings);

			var _command = new JsonObject()
			{
				{ "/layoutset/update-settings", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task UpdateLayoutSetPrototypeLinkEnabledAsync(long groupId, bool privateLayout, bool layoutSetPrototypeLinkEnabled, string layoutSetPrototypeUuid)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("privateLayout", privateLayout);
			_parameters.Add("layoutSetPrototypeLinkEnabled", layoutSetPrototypeLinkEnabled);
			_parameters.Add("layoutSetPrototypeUuid", layoutSetPrototypeUuid);

			var _command = new JsonObject()
			{
				{ "/layoutset/update-layout-set-prototype-link-enabled", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task UpdateLogoAsync(long groupId, bool privateLayout, bool logo, byte[] bytes)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("privateLayout", privateLayout);
			_parameters.Add("logo", logo);
			_parameters.Add("bytes", bytes);

			var _command = new JsonObject()
			{
				{ "/layoutset/update-logo", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task UpdateLogoAsync(long groupId, bool privateLayout, bool logo, Stream file)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("privateLayout", privateLayout);
			_parameters.Add("logo", logo);
			_parameters.Add("file", file);

			var _command = new JsonObject()
			{
				{ "/layoutset/update-logo", _parameters }
			};

			await this.Session.UploadAsync(_command);
		}

		public async Task<dynamic> UpdateVirtualHostAsync(long groupId, bool privateLayout, string virtualHost)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("privateLayout", privateLayout);
			_parameters.Add("virtualHost", virtualHost);

			var _command = new JsonObject()
			{
				{ "/layoutset/update-virtual-host", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/layoutset/LayoutSetService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.layoutset;

import com.liferay.mobile.android.http.file.UploadData;
import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class LayoutSetService extends BaseService {

	public LayoutSetService(Session session) {
		super(session);
	}

	public void updateLayoutSetPrototypeLinkEnabled(long groupId, boolean privateLayout, boolean layoutSetPrototypeLinkEnabled, String layoutSetPrototypeUuid) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("layoutSetPrototypeLinkEnabled", layoutSetPrototypeLinkEnabled);
			_params.put("layoutSetPrototypeUuid", checkNull(layoutSetPrototypeUuid));
			_params.put("privateLayout", privateLayout);

			_command.put("/layoutset/update-layout-set-prototype-link-enabled", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public void updateLogo(long groupId, boolean privateLayout, boolean hasLogo, byte[] bytes) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("bytes", toString(bytes));
			_params.put("groupId", groupId);
			_params.put("hasLogo", hasLogo);
			_params.put("privateLayout", privateLayout);

			_command.put("/layoutset/update-logo", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public void updateLogo(long groupId, boolean privateLayout, boolean hasLogo, UploadData file) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("file", checkNull(file));
			_params.put("groupId", groupId);
			_params.put("hasLogo", hasLogo);
			_params.put("privateLayout", privateLayout);

			_command.put("/layoutset/update-logo", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.upload(_command);
	}

	public JSONObject updateLookAndFeel(long groupId, boolean privateLayout, String themeId, String colorSchemeId, String css) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("colorSchemeId", checkNull(colorSchemeId));
			_params.put("css", checkNull(css));
			_params.put("groupId", groupId);
			_params.put("privateLayout", privateLayout);
			_params.put("themeId", checkNull(themeId));

			_command.put("/layoutset/update-look-and-feel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateSettings(long groupId, boolean privateLayout, String settings) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("privateLayout", privateLayout);
			_params.put("settings", checkNull(settings));

			_command.put("/layoutset/update-settings", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateVirtualHost(long groupId, boolean privateLayout, String virtualHostname) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("privateLayout", privateLayout);
			_params.put("virtualHostname", checkNull(virtualHostname));

			_command.put("/layoutset/update-virtual-host", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateVirtualHosts(long groupId, boolean privateLayout, JSONObject virtualHostnames) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);
			_params.put("privateLayout", privateLayout);
			_params.put("virtualHostnames", checkNull(virtualHostnames));

			_command.put("/layoutset/update-virtual-hosts", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}